layers/object_tracker/object_lifetime_validation.h

```diff
@@ -133,52 +133,65 @@ class ObjectLifetimes : public ValidationObject {
+            if (lifetimes && lifetimes != this && lifetimes->TracksObject(object_handle, object_type)) {
+                other_lifetimes = lifetimes;
                 break;
             }
         }
-        // Report about findings.
-        if (!found) {
-            // Object was not found anywhere
-            return LogError(invalid_handle_code, instance, loc, "Invalid %s Object 0x%" PRIxLEAST64 ".", object_string[object_type],
+        // Object was not found anywhere
+        if (!other_lifetimes) {
+            return LogError(invalid_handle_vuid, instance, loc, "Invalid %s Object 0x%" PRIxLEAST64 ".", object_string[object_type],
                             object_handle);
-        } else if (wrong_device_code != kVUIDUndefined && object_type != kVulkanObjectTypeSurfaceKHR) {
-            // Object found on other device and device error code is provided
-            const LogObjectList objlist(instance, device, other_device);
-            return LogError(wrong_device_code, objlist, loc,
-                            "Expected all Dispatchable Handles to use %s, but the %s (0x%" PRIxLEAST64
-                            ") was created, allocated or retrieved from %s.",
-                            FormatHandle(device).c_str(), object_string[object_type], object_handle,
-                            FormatHandle(other_device).c_str());
+        }
+        // Anonymous object validation does not check parent, only that the object exists
+        if (wrong_parent_vuid == kVUIDUndefined) {
+            return skip;
+        }
+        // Object found on another device
+        LogObjectList objlist;
+        std::string handle_str;
+        std::string other_handle_str;
+        if (parent_type == kVulkanObjectTypeDevice) {
+            objlist = LogObjectList(instance, device, other_lifetimes->device);
+            handle_str = FormatHandle(device);
+            other_handle_str = FormatHandle(other_lifetimes->device);
+        } else if (parent_type == kVulkanObjectTypeInstance) {
+            objlist = LogObjectList(instance, other_lifetimes->instance);
+            handle_str = FormatHandle(instance);
+            other_handle_str = FormatHandle(other_lifetimes->instance);
+        } else if (parent_type == kVulkanObjectTypePhysicalDevice) {
+            objlist = LogObjectList(instance, physical_device, other_lifetimes->physical_device);
+            handle_str = FormatHandle(physical_device);
+            other_handle_str = FormatHandle(other_lifetimes->physical_device);
         } else {
-            // Object was found on other device, but it's not a device error.
-            // TODO: this could be an instance-level error, we need to add support for this.
+            assert(false);
             return skip;
         }
+        return LogError(wrong_parent_vuid, objlist, loc,
+                        "%s (0x%" PRIxLEAST64 ") was not created, allocated or retrieved from %s, but from another %s",
```

Comment:
I think would better to make clear which is used and which "should" be used

also change
`vkCmdFillBuffer(): dstBuffer VkBuffer (0xfa21a40000000003) ...`
to 
`vkCmdFillBuffer(): dstBuffer (VkBuffer 0xfa21a40000000003) ...`

```suggestion
"(%s 0x%" PRIxLEAST64 ") was created, allocated or retrieved from %s, but command is using %s",
```

Reply:
Some of the VUIDs refer to a parent object that is not a command's parameter but is in the parent chain of the dispatchable parameter (e.g. `VUID-vkGetDeviceGroupSurfacePresentModesKHR-commonparent`, `VUID-vkGetPhysicalDeviceSurfaceSupportKHR-commonparent`). I extended the second part of the message to this:
`but command is using (or its dispatchable parameter is associated with) %s`

Let me know if that's not too confusing. We probably also can detect these two cases in codegen and generate separate messages if needed.

Reply:
I like the new message